src/configuration.cpp

```diff
@@ -0,0 +1,702 @@
+                            {
+                                value = loc;
+                                return;
+                            }
+                        }
+
+                        LOG_ERROR << "ssl_verify is enabled but no ca certificates found";
+                        exit(1);
+                    }
+                    else
+                    {
+                        value = "<system>";
+                        return;
+                    }
+                }
+            }
+        };
+
+        void always_link_hook(bool&)
+        {
+            auto& config = Configuration::instance();
+            auto& always_softlink = config.at("always_softlink")
+                                        .compute_config(ConfigurationLevel::kFile, false)
+                                        .value<bool>();
+            auto& always_copy = config.at("always_copy")
+                                    .compute_config(ConfigurationLevel::kFile, false)
+                                    .value<bool>();
+
+            if (always_softlink && always_copy)
+            {
+                throw std::runtime_error(
+                    "'always-softlink' and 'always-copy' options are exclusive.");
+            }
+        }
+    }
+
+    Configuration::Configuration()
+    {
+        set_configurables();
+    }
+
+    Configuration& Configuration::instance()
+    {
+        static Configuration config;
+        return config;
+    }
+
+    void Configuration::set_configurables()
+    {
+        auto& ctx = Context::instance();
+
+        // Channels
+        insert(Configurable("channels", &ctx.channels)
+                   .group("Channels")
+                   .set_env_var_name("CONDA_CHANNELS")
+                   .description("Define the list of channels")
+                   .long_description(unindent(R"(
+                    The list of channels where the packages will be search for.
+                    See also 'channel_priority'.)")));
+
+        insert(Configurable("channel_alias", &ctx.channel_alias)
+                   .group("Channels")
+                   .set_env_var_name()
+                   .description("The prepended url location to associate with channel names"));
+
+        insert(Configurable("default_channels", &ctx.default_channels)
+                   .group("Channels")
+                   .set_env_var_name()
+                   .description("Default channels used")
+                   .long_description(unindent(R"(
+                    The list of channel names and/or urls used for the 'defaults'
+                    multichannel.)")));
+
+        insert(Configurable("override_channels_enabled", &ctx.override_channels_enabled)
+                   .group("Channels")
+                   .set_env_var_name()
+                   .description("Permit use of the --overide-channels command-line flag"));
+
+        // Network
+        insert(Configurable("local_repodata_ttl", &ctx.local_repodata_ttl)
```

Comment:
did you rename this? I think I had a flag for this before, but unsure what name I used.

Reply:
the corresponding CLI flag is `--repodata-ttl`